src/librustc/middle/ty/context.rs

```diff
@@ -1000,4 +1030,38 @@ impl<'tcx> ctxt<'tcx> {
     pub fn mk_param_from_def(&self, def: &ty::TypeParameterDef) -> Ty<'tcx> {
         self.mk_param(def.space, def.index, def.name)
     }
+
+    pub fn trait_items(&self, trait_did: DefId) -> Rc<Vec<ty::ImplOrTraitItem<'tcx>>> {
+        // since this is cached, pushing a dep-node for the
+        // computation yields the correct dependencies.
```

Comment:
@michaelwoerister 

> Isn't this the case that DepTrackingMap::memoize() is for?

Ah yes, I meant to port this and `repr_hints` to use the new `memoize()` helper, but forgot.